### PR TITLE
TRUNK-5683: getConceptsByClass(ConceptClass) to use Hibernate Criteria.

### PR DESCRIPTION
**Pull request by samuelmale**
Ticket: https://issues.openmrs.org/browse/TRUNK-5683

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/ConceptServiceTest.java [hunk 1]
```diff
@@ -2245,9 +2245,30 @@ public void getConceptsByAnswer_shouldFindAnswersForConcept() {
...
-		cs.getConceptsByClass(cs.getConceptClass(1));
+	public void getConceptsByClass_shouldGetConceptsByClass() {
+		// setup
+		ConceptClass cc = new ConceptClass(3);
+		
+		// replay
+		List<Concept> concepts = conceptService.getConceptsByClass(cc);
+		
+		// verify
+		Assert.assertEquals(4, concepts.size());
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: Can you verify that you got the 4 concepts that you expected?

Comment (mogoodrich) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: +1

Comment (samuelmale) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: Nice catch @mks-d, just worked on that!

File: api/src/test/java/org/openmrs/api/ConceptServiceTest.java [hunk 2]
```diff
@@ -2245,9 +2245,40 @@ public void getConceptsByAnswer_shouldFindAnswersForConcept() {
...
+	public void getConceptsByClass_shouldReturnAnEpmtyListIfNoneWasFound() {
+		// setup
+		ConceptClass cc = new ConceptClass(23);
+		
+		// replay
+		List<Concept> concepts = conceptService.getConceptsByClass(cc);
+		
+		// verify
+		Assert.assertTrue(concepts.isEmpty());
 	}
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: Please do this:
```java
/**
 * @see ConceptService#getConceptsByClass(ConceptClass)
 */
@Test
public void getConceptsByClass_shouldGetUnorderedConceptsForClass() {
  // replay
  List<Concept> actualConcepts = conceptService.getConceptsByClass(new ConceptClass(3));
  
  // verify
  Assert.assertThat(actualConcepts.size(), is(4));
  Assert.assertThat(actualConcepts, containsInAnyOrder(conceptService.getConcept(3), conceptService.getConcept(60), conceptService.getConcept(88), conceptService.getConcept(792)));
}

/**
 * @see ConceptService#getConceptsByClass(ConceptClass)
 */
@Test
public void getConceptsByClass_shouldReturnEmptyListForEmptyClass() {
  // replay
  List<Concept> concepts = conceptService.getConceptsByClass(new ConceptClass(23));
  
  // verify
  Assert.assertTrue(concepts.isEmpty());
}
```
It is really not ok to decouple your test method by resetting the UUIDs _again_, even though they are defined already in **standardTestDataset.xml**.

Furthermore your test did not resist reordering the loading order in **standardTestDataset.xml**.

Finally there was a typo in the second method name: `AnEpmtyList`, I suggested new names anyway.

Comment (samuelmale) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: Thanks for the advice.. actually this makes sense and looks way better

Comment (mks-d) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: @samuelmale can you apply the changes on a subsequent commit?

File: api/src/test/java/org/openmrs/api/ConceptServiceTest.java [hunk 3]
```diff
@@ -2245,9 +2245,27 @@ public void getConceptsByAnswer_shouldFindAnswersForConcept() {
...
 	@Test
-	public void getConceptsByClass_shouldNotFailDueToNoNameInSearch() {
-		ConceptService cs = Context.getConceptService();
-		cs.getConceptsByClass(cs.getConceptClass(1));
+	public void getConceptsByClass_shouldGetConceptsByClass() {				
+		// replay
+		List<Concept> actualConcepts = conceptService.getConceptsByClass(new ConceptClass(3));
+		
+		// verify
+		Assert.assertThat(actualConcepts, containsInAnyOrder(conceptService.getConcept(3), conceptService.getConcept(60), conceptService.getConcept(88), conceptService.getConcept(792)));
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: You need to assert that the size is exactly 4, I did specify it in my last review:
```java
Assert.assertThat(actualConcepts.size(), is(4));
```

Comment (samuelmale) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java]: How did I miss this! Oohh I'm sorry.

Let me update it!

**Pull request merged**